SourceX/controls/plrctrls.cpp

```diff
@@ -0,0 +1,693 @@
+#include "controls/plrctrls.h"
+

```

Comment:
All the time based code in this function makes me think it was written to handle input by periodically pooling the button state rather then handling events, maybe this can be cleaned up a bit.